lib/rules/require-dollar-before-jquery-assignment.js

```diff
@@ -75,6 +75,41 @@ module.exports.prototype = {
             var varName;
             var right;
 
+            function checkIfVarNameShouldStartWithDollar(left, varName, right) {
+
+                if (varName.indexOf('$') === 0 || varName.indexOf('_$') === 0) {
+                    return;
+                }
+
+                if (!right || right.type !== 'CallExpression') {
+                    return;
+                }
+
+                var nextToken = file.getTokenByRangeStart(right.callee.range[0]);
+                if (nextToken.value !== '$') {
+                    return;
+                }
+
+                nextToken = file.getNextToken(nextToken);
+                if (nextToken.value !== '(') {
+                    return;
+                }
+
+                while (!(nextToken.type === 'Punctuator' && nextToken.value === ')')) {
+                    nextToken = file.getNextToken(nextToken);
+                }
+
+                nextToken = file.getNextToken(nextToken);
+
+                if (!nextToken || !(nextToken.type === 'Punctuator' && nextToken.value === '.')) {
```

Comment:
I can't think of a condition where it get's to this point with ``` !nextToken ``` being true. If you can give me a clue, I'm happy to add the test =)